tools/dataset_converters/tier4dataset_converter.py

```diff
@@ -0,0 +1,650 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+from collections import OrderedDict
+from os import path as osp
+from typing import List, Tuple, Union
+
+import mmcv
+import mmengine
+import numpy as np
+import yaml
+from nuscenes.nuscenes import NuScenes
+from nuscenes.utils.geometry_utils import view_points
+from pyquaternion import Quaternion
+from shapely.geometry import MultiPoint, box
+
+from mmdet3d.datasets.convert_utils import NuScenesNameMapping
+from mmdet3d.structures import points_cam2img
+
+nus_categories = ('car', 'truck', 'bus', 'bicycle', 'pedestrian')
+
+nus_attributes = ('cycle.with_rider', 'cycle.without_rider',
+                  'pedestrian.moving', 'pedestrian.standing',
+                  'pedestrian.sitting_lying_down', 'vehicle.moving',
+                  'vehicle.parked', 'vehicle.stopped', 'None')
+
+
+def create_tier4_infos(root_path,
+                       info_prefix,
+                       version='sample_dataset',
+                       max_sweeps=0):
+    """Create info file of tier4 dataset.
+
+    Given the raw data, generate its related info file in pkl format.
+
+    Args:
+        root_path (str): Path of the data root.
+        info_prefix (str): Prefix of the info file to be generated.
+        version (str, optional): Version of the data.
+        max_sweeps (int, optional): Max number of sweeps.
+            Default: 1.
+    """
+
+    def process_scene(nusc, train_scenes, val_scenes, is_test=False):
+        if not is_test:
+            train_nusc_infos, val_nusc_infos = _fill_trainval_infos(
+                nusc, train_scenes, val_scenes, False, max_sweeps=max_sweeps)
+            return train_nusc_infos, val_nusc_infos
+        else:
+            train_nusc_infos, _ = _fill_trainval_infos(
+                nusc, train_scenes, val_scenes, True, max_sweeps=max_sweeps)
+            return train_nusc_infos
+
+    dataset_config = root_path + version + '.yaml'
```

Comment:
@beginningfan with which command did you get this error? T4Dataset is the dataset format, and sample_dataset is a version in there.

Reply:
@kaancolak Oh, my fault. I forgot to click on `review changes` before, which caused the comment to remain in a pending state。
I used default version param, which is v1.0. It will be ok when using `sample_dataset` for the version. Then I suggest writing the T4Dataset version in the document, or modifying the default parameters of the T4Dataset version number.